package frc.robot.commands.autonomous;

import com.pathplanner.lib.auto.AutoBuilder;
import com.pathplanner.lib.path.PathPlannerPath;
import org.wpilib.math.geometry.Pose2d;
import org.wpilib.driverstation.DriverStationErrors;
import org.wpilib.framework.RobotBase;
import org.wpilib.command2.Commands;
import org.wpilib.command2.SequentialCommandGroup;
import frc.robot.subsystems.groundIntakeLinearExtension.GroundIntakeLinearExtensionSubsystem;
import frc.robot.subsystems.groundIntakeRoller.GroundIntakeRollerSubsystem;
import frc.robot.subsystems.indexer.IndexerSubsystem;
import frc.robot.subsystems.shooter.ShooterSubsystem;

public class DelayAuto extends SequentialCommandGroup {
  public DelayAuto(
      ShooterSubsystem shooter,
      IndexerSubsystem indexer,
      GroundIntakeLinearExtensionSubsystem groundIntakeExtend,
      GroundIntakeRollerSubsystem groundIntakeRoller) {

    try {
      PathPlannerPath path = PathPlannerPath.fromChoreoTrajectory("Delay_Start_Center");
      // PathPlannerPath another_path = PathPlannerPath.fromChoreoTrajectory("PATH NAME");

      Pose2d startingPose =
          new Pose2d(path.getPoint(0).position, path.getIdealStartingState().rotation());

      addCommands(
          AutoBuilder.resetOdom(startingPose).onlyIf(() -> RobotBase.isSimulation()),
          Commands.waitSeconds(5).deadlineFor(indexer.index(), shooter.runVelocity(45, 42)),
          Commands.waitSeconds(9)
              .deadlineFor(Commands.parallel(shooter.runVoltage(0), indexer.runVoltage(0))),
          Commands.parallel(
              AutoBuilder.followPath(path),
              Commands.sequence(
                  Commands.waitSeconds(2.5),
                  Commands.parallel(
                      groundIntakeExtend.extend2(), groundIntakeRoller.startIntake()))));

    } catch (Exception e) {
      DriverStationErrors.reportError("Path Not Found: " + e.getMessage(), e.getStackTrace());
    }
  }
}
